src/System.Net.NameResolution/src/System/Net/NameResolutionPal.Unix.cs

```diff
@@ -38,90 +44,84 @@ private static SocketError GetSocketErrorForNativeError(int error)
             }
         }
 
-        private static unsafe IPHostEntry CreateIPHostEntry(Interop.Sys.HostEntry hostEntry)
+        private static unsafe void ParseHostEntry(Interop.Sys.HostEntry hostEntry, bool justAddresses, out string hostName, out string[] aliases, out IPAddress[] addresses)
         {
-            string hostName = null;
-            if (hostEntry.CanonicalName != null)
+            try
```

Comment:
> It doesn't look like there are any throws or early exits.

We allocate a bunch.  I wanted to make sure that we free'd the native allocation and avoid a leak in the event that we hit a temporary OOM, in case the caller tries to recover.